src/os_posix/os_fs.c

```diff
@@ -1068,7 +1070,7 @@ __wt_prepare_remap_resize_file(WT_FILE_HANDLE *file_handle, WT_SESSION *wt_sessi
     sleep_usec = 10;
     yield_count = 0;
 
-    if (!pfh->mmap_file_mappable)
+    if (pfh->mmap_buf == NULL)
```

Comment:
Should this be an assertion? Resizing an unmapped file looks like a bug to my eyes.

Reply:
Possibly? I think it should be another ticket though, I'm trying not to change behaviour here and `__posix_file_truncate` expects to be able to call this on any old file handle.